core/domain/exp_domain.py

```diff
@@ -4118,16 +4154,31 @@ def _convert_v28_dict_to_v29_dict(cls, exploration_dict):
 
         return exploration_dict
 
+    @classmethod
+    def _convert_v29_dict_to_v30_dict(cls, exploration_dict):
+        """Converts a v29 exploration dict into a v30 exploration dict.
+
+        Adds dimensions to all oppia-noninteractive-image tags.
+        """
+        exploration_dict['schema_version'] = 30
+
+        exploration_dict['states'] = cls._convert_states_v24_dict_to_v25_dict(
+            exploration_dict['id'], exploration_dict['states'])
+        exploration_dict['states_schema_version'] = 25
+
+        return exploration_dict
+
     @classmethod
     def _migrate_to_latest_yaml_version(
-            cls, yaml_content, title=None, category=None):
+            cls, yaml_content, exp_id, title=None, category=None):
         """Return the YAML content of the exploration in the latest schema
         format.
 
         Args:
             yaml_content: str. The YAML representation of the exploration.
             title: str. The exploration title.
             category: str. The exploration category.
+            exp_id: str. ID of the exploration.
```

Comment:
Order of args should match order in docstring.

Reply:
Done